dotnet/src/Microsoft.Agents.AI.Abstractions.CosmosNoSql/CosmosChatMessageStore.cs

```diff
@@ -0,0 +1,853 @@
+        this._ownsClient = false;
+
+        if (string.IsNullOrWhiteSpace(databaseId))
+        {
+            throw new ArgumentException("Cannot be null or whitespace", nameof(databaseId));
+        }
+
+        if (string.IsNullOrWhiteSpace(containerId))
+        {
+            throw new ArgumentException("Cannot be null or whitespace", nameof(containerId));
+        }
+
+        if (string.IsNullOrWhiteSpace(tenantId))
+        {
+            throw new ArgumentException("Tenant ID cannot be null or whitespace.", nameof(tenantId));
+        }
+
+        if (string.IsNullOrWhiteSpace(userId))
+        {
+            throw new ArgumentException("User ID cannot be null or whitespace.", nameof(userId));
+        }
+
+        if (string.IsNullOrWhiteSpace(sessionId))
+        {
+            throw new ArgumentException("Session ID cannot be null or whitespace.", nameof(sessionId));
+        }
+
+        this._container = this._cosmosClient.GetContainer(databaseId, containerId);
+        this._conversationId = sessionId; // Use sessionId as conversationId for compatibility
+        this._databaseId = databaseId;
+        this._containerId = containerId;
+
+        // Initialize hierarchical partitioning mode
+        this._tenantId = tenantId;
+        this._userId = userId;
+        this._useHierarchicalPartitioning = true;
+        // Use native hierarchical partition key with PartitionKeyBuilder
+        this._partitionKey = new PartitionKeyBuilder()
+            .Add(tenantId)
+            .Add(userId)
+            .Add(sessionId)
+            .Build();
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CosmosChatMessageStore"/> class from previously serialized state.
+    /// </summary>
+    /// <param name="serializedStoreState">A <see cref="JsonElement"/> representing the serialized state of the message store.</param>
+    /// <param name="cosmosClient">The <see cref="CosmosClient"/> instance to use for Cosmos DB operations.</param>
+    /// <param name="jsonSerializerOptions">Optional settings for customizing the JSON deserialization process.</param>
+    /// <exception cref="ArgumentNullException">Thrown when <paramref name="cosmosClient"/> is null.</exception>
+    /// <exception cref="ArgumentException">Thrown when the serialized state cannot be deserialized.</exception>
+    [UnconditionalSuppressMessage("Trimming", "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access", Justification = "StoreState type is controlled and used for serialization")]
+    [UnconditionalSuppressMessage("AOT", "IL3050:Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling", Justification = "StoreState type is controlled and used for serialization")]
+    public CosmosChatMessageStore(JsonElement serializedStoreState, CosmosClient cosmosClient, JsonSerializerOptions? jsonSerializerOptions = null)
```

Comment:
For other constructors that take CosmosClient it typically comes first in the param list.  Have you considered making it the same here?

```suggestion
    public CosmosChatMessageStore(CosmosClient cosmosClient, JsonElement serializedStoreState, JsonSerializerOptions? jsonSerializerOptions = null)
```